### ISSUE TITLE
Difficulties with deploying the plugin

### ISSUE THREAD

**Issue by JJTofflemire** [andrewbrereton/obsidian-to-ical-plugin#29]
Hey! I followed your instructions from the README.md and from [this issue](https://github.com/andrewbrereton/obsidian-to-ical-plugin/issues/22), but I still haven't had any luck with getting the plugin to work. I have filled out all the sections in both the gist and local saving sections, but can't get either to update. I have filled out every section in both the gist and local sections. Here are the tasks I am trying to get recognized (in a `To Do.md` file):

```
- [ ] #task 2023-12-04 work on stuff

- [ ] 2023-12-05 tasks
```

I am running Obsidian on Macos 14. What other information do I need to provide?

![Screenshot 2023-12-04 at 11 42 06 AM](https://github.com/andrewbrereton/obsidian-to-ical-plugin/assets/85974663/443bceff-5e4c-4cd0-8a9c-20f67c71cc27)

One interesting note is that every time I paste in my github personal access key, the field gets cleared. I'm not sure if this is intentional or part of the problem.

**Comment by JJTofflemire** [comment]
I figured that my problem was that I was manually making tasks instead of using the official Obsidian Tasks plugin with due dates and scheduled dates etc, but still no dice.

**Comment by JJTofflemire** [comment]
Yes, I have migrated over to using the standard Tasks format with dates now. (Don't know how it took me this long to find out about Tasks)

**Comment by andrewbrereton** [comment]
Hi @JJTofflemire, thanks for raising this issue.

```markdown
- [ ] #task 2023-12-04 work on stuff
- [ ] 2023-12-05 tasks
```

This should absolutely be identified as a Task. The regular expression to find a date in a line is:

```
/(?<emoji>➕|⏳|🛫|📅|✅)?\s?(?<year>\d{4})-(?<month>\d{2})-(?<day>\d{1,2})\b/gi
```

The `<emoji>` group is optional. A long as there is a "0000-00-00" or "0000-00-0" where 0 is any number, then it should be considered a date and therefore a valid task with a date.

In my test case I have:

<img width="339" alt="image" src="https://github.com/andrewbrereton/obsidian-to-ical-plugin/assets/682893/eedce9c3-4655-4134-b30b-4b39b7fc6413">

And in the console I can see it was identified correctly:

<img width="576" alt="image" src="https://github.com/andrewbrereton/obsidian-to-ical-plugin/assets/682893/833350f2-54ac-497f-82a9-d361e8dfe262">

So then I added your test case:

<img width="340" alt="image" src="https://github.com/andrewbrereton/obsidian-to-ical-plugin/assets/682893/8f3e7a54-374c-4178-ae71-437cce2d5855">

And it worked 😳:

<img width="582" alt="image" src="https://github.com/andrewbrereton/obsidian-to-ical-plugin/assets/682893/13c763bc-c19c-4f10-88e4-bfb8e32a383a">

I'm at a loss trying to understand how this could happen. I've added some additional logging and a new setting called "Debug mode" to print more logs to console. Perhaps that will assist me in working out what is going on here.

**Comment by JJTofflemire** [comment]
That's really weird. I configured the plugin on my desktop running Arch Linux, and still no dice. Here's the output from `Debug Mode`.

![image](https://github.com/andrewbrereton/obsidian-to-ical-plugin/assets/85974663/2d5aa266-c39c-4786-8cb8-e860799173a4)

Which failed to find these tasks:

```
- [ ] Module 1 and 2 review 📅 2023-12-08 ⏳ 2023-12-07 
- [ ] Section 17.2 # 1, 5, 15 p 979 🔼 ⏳ 2023-12-05
- [ ] Webwork Set 23 (4 problems) ⏫ ⏳ 2023-12-05 📅 2023-12-06
- [ ] Homework 11 📅 2023-12-11 ⏬ 
```

The link from the iCal settings successfully pulls up my gist, though there is still an error in the console. Also, when I tried to enter my Github token, every time I tried to type in the box it would immediately deselect (and pasting in the token would give me no feedback). Though, regardless, the primary issue appears to be identifying the tasks.

**Comment by andrewbrereton** [comment]
I'm wondering about the GitHub token. Can you confirm the details of the token?

I currently enforce the token to be either a classic token or a fine-grained token like this:

```
^ghp_[a-zA-Z0-9]{255}$
```

or

```
^github_pat_[a-zA-Z0-9]{22}_[a-zA-Z0-9]{59}$
```

Can you check if your token looks like this or if it looks different? If it's different, then the code will throw an error. That would explain the token being cleared, and maybe even the plugin not working correctly from there on.

**Comment by JJTofflemire** [comment]
Yes, that was the problem! I switched to a fine-grained access token, and now the plugin works as intended (both writing to a gist and to a .ical file locally). My classic token was only 40 characters long, which was causing all the problems. I'm not sure if I was failing to copy the whole token or if they are not always 255 characters long

**Comment by andrewbrereton** [comment]
That's great to hear. I will update the code to be a bit better about what should happen when there is a problem parsing the token. eg: Tell the user instead of simply clearing the textbox.

**Comment by andrewbrereton** [comment]
This has been fixed in [1.9.1](https://github.com/andrewbrereton/obsidian-to-ical-plugin/releases/tag/1.9.2)
